infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/RustKZG.java

```diff
@@ -89,18 +95,44 @@ public boolean verifyBlobKzgProofBatch(
       final List<KZGCommitment> kzgCommitments,
       final List<KZGProof> kzgProofs)
       throws KZGException {
-    throw new RuntimeException("LibPeerDASKZG library doesn't support verifyBlobKzgProofBatch");
+    try {
+      final byte[][] blobsBytes = new byte[blobs.size()][];
+      IntStream.range(0, blobs.size())
+          .forEach(index -> blobsBytes[index] = blobs.get(index).toArrayUnsafe());
+      final byte[][] commitmentsBytes = new byte[kzgCommitments.size()][];
+      IntStream.range(0, kzgCommitments.size())
+          .forEach(index -> commitmentsBytes[index] = kzgCommitments.get(index).toArrayUnsafe());
+      final byte[][] proofsBytes = new byte[kzgProofs.size()][];
+      IntStream.range(0, kzgProofs.size())
+          .forEach(index -> proofsBytes[index] = kzgProofs.get(index).toArrayUnsafe());
```

Comment:
this would likely be cleaner if we had 1 for loop and 3 calls?
```
for (int i = 0; i < blobs.size(); i++) {
  blobsBytes[i] = blobs.get(i).toArrayUnsafe();
  commitmentsBytes[i] = kzgCommitments.get(i).toArrayUnsafe();
  proofsBytes[i] = kzgProofs.get(i).toArrayUnsafe();
}
```

Reply:
Ahh yeah, it's deneb, I'm used we have more proofs in PeerDAS